### ISSUE TITLE
Wrong value recorded. onChange() callback is not fired when the new input date is in invalid format

### ISSUE THREAD

**Issue by twlai** [arthurdenner/react-semantic-ui-datepickers#334]
# 🐛 bug report

## Description of the problem
Under the following situation, there's a problem when using the onChange() callback to update the value to state variable. 

1. Focus the text input box
2. The date picker is popped up. Select a date. 
3. The selected date value is displayed in the input text box.
4. Then focus the date input box again.
5. This time manually use backspace to remove some digits, resulting a invalid date format
6. Then move the focus away from the date input text box. 
7. The invalid date is removed from the date input box, resulting an empty string being shown in the date input text box. 

However, step (6) or (7) does not fire another onChange() call. 
In other words, the state variable has no way to be updated, and is still having the value after step (3) or (4). 

## How has this issue affected you? What are you trying to accomplish?
This can result a wrong value is recorded and submitted. 

Apparently, step (6) or (7) should fire a proper onChange() call so the developer using the library can update their state variables too.

**Comment by arthurdenner** [comment]
Hi, @twlai! 👋 

Thank you for reporting this.

I've created a PR to fix it and released `2.8.1-beta.0` so that you can test if it works for you.

Let me know how it goes please 😃

**Comment by arthurdenner** [comment]
@twlai, have you been able to test the fix?

**Comment by twlai** [comment]
Dear Arthur,

Sorry for getting back late. I have just tested the fix. It works like a
charm. Thank you so much.

Regards,
TW
